### PR TITLE
Update folder structure for components

### PR DESCRIPTION
**Pull request by matthijsgroen**

### CODE REVIEW

**Comment by matthijsgroen** [comment]
@pascalw I asked for your review specifically for the versioning and contribution setup

**Review by ljpengelen — APPROVED** [review]
Looks fine to me.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+The components should add as little extra dependencies to the project as
+possible, unless the component is a direct usage of that dependency.
+(e.g. local forage drivers depending on local forage)
+
+Currenty we have the following categories to contribute to:
+
+- Factories. Using the `js-factories` npm lib, the factories should be
+  general purpose (backbone/marionette/cordova) and never be domain
+  specific.
+- Components. These should be stand alone instances in your app.
```
Comment (pascalw) [inline CONTRIBUTING.md:19]: This is a bit vague really. Maybe add a few examples to make it a bit more concrete?

File: README.md [hunk 2]
```diff
@@ -20,7 +20,29 @@ You'll never use all of maji-extras, but you pick exactly the parts that you nee
...
+  - Directory structure changes
+  - File names of components change
+* Update of Minor when:
+  - New components are added
+* Update of Patch when:
+  - Components are updated
+
+Components should already be stable before they are added to
+Maji-extras. When a component is radically changed and would break the
+interface, it should be added as a new component under a different name.
```
Comment (pascalw) [inline README.md:47]: Hm that's not great. Breaking changes can be very small and I think our process should support it.

File: src/components/nativestorage_driver.coffee [hunk 3]
```diff
@@ -25,15 +25,18 @@ _initStorage = (options) ->
         .catch (error) ->
           _setKeys([], dbInfo)
 
+_patchPromise = (promise) ->
+  promise.catch ?= promise.fail
```
Comment (pascalw) [inline src/components/nativestorage_driver.coffee:29]: Maybe add a comment here describing why we do this

Comment (pascalw) [reply src/components/nativestorage_driver.coffee:29]: Btw I would prefer if these changes go into a separate PR as they are unrelated to the structure of this lib.

File: README.md (hunk 2 above)
Comment (matthijsgroen) [reply README.md:47]: What would you propose?

File: src/components/nativestorage_driver.coffee (hunk 3 above)
Comment (matthijsgroen) [reply src/components/nativestorage_driver.coffee:29]: I will split this up

**Comment by matthijsgroen** [comment]
Deconstructed into:

- #10 -- The plain rename
- #11 -- The versioning strategy
- #12 -- The contribution guideline
- #13 -- Fixing of promises / return values of nativestorage

**Pull request closed without merging**